packages/webapp/src/components/Task/AddSoilAmendmentProducts/QuantityApplicationRate/index.tsx

```diff
@@ -145,7 +139,7 @@ const QuantityApplicationRate = ({
         leftLabel={formatLabel('ADD_TASK.SOIL_AMENDMENT_VIEW.WEIGHT', isWeight)}
         label={formatLabel('ADD_TASK.SOIL_AMENDMENT_VIEW.VOLUME', !isWeight)}
         isToggleVariant
-        onChange={toggleMeasure}
+        onChange={(e) => setValue(IS_WEIGHT, e.target.checked ? false : true)}
```

Comment:
I think the previous way of doing it was correct: `setIsWeight((prev) => !prev);`

Reply:
It isn't component state anymore so it doesn't have that kind of setter (just the React Hook Form `setValue()` setter).